### PR TITLE
Remove UA farbling on all platforms and unconditionally exclude Android model

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/15372
Resolves https://github.com/brave/brave-browser/issues/15322

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/farbling/brave_navigator_useragent_farbling_browsertest.cc [hunk 1]
```diff
@@ -140,17 +111,11 @@ IN_PROC_BROWSER_TEST_F(BraveNavigatorUserAgentFarblingBrowserTest,
   // Farbling level: off
   AllowFingerprinting(domain_b);
   NavigateToURLUntilLoadStop(url_b);
-  // HTTP User-Agent header we just sent in that request should be the same as
-  // the unfarbled user agent
-  EXPECT_EQ(last_requested_http_user_agent(), unfarbled_ua);
```
Comment (bridiver) [inline browser/farbling/brave_navigator_useragent_farbling_browsertest.cc:145]: I don't think we should be removing these checks. We still want to verify that the network user agent is not farbled in all cases

**Review by bridiver — APPROVED** [review]
please add network UA tests back in always checking against the unfarbled UA

File: browser/farbling/brave_navigator_useragent_farbling_browsertest.cc (hunk 1 above)
Comment (pilgrim-brave) [reply browser/farbling/brave_navigator_useragent_farbling_browsertest.cc]: fixed in https://github.com/brave/brave-core/pull/8562/commits/880e03d5e7093cb95ee0e0d50120e142ae34a246

**Comment by pilgrim-brave** [comment]
Added new DEPS file since review, as noted by gn_check

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.25.33 Chromium: 90.0.4430.85 (Official Build) nightly (64-bit)
-- | --
Revision | 5bc145d831c180d9ff94f29a0d7a2e1cbd30ef36-refs/branch-heads/4430@{#1311}
OS | Windows 10 OS Version 2009 (Build 19042.928)
```

### Test Case https://github.com/brave/brave-browser/issues/15039

Went through the STR/Cases outlined via https://github.com/brave/brave-browser/issues/15039#issue-846336230 and ensured that Cloudflare correctly directed/opened http://chaingames.io.

**`Desktop`** | **`Android`**
---------------|-----------------
image | Screenshot_20210421-223946_Brave - Nightly

### Test Case https://github.com/brave/brave-browser/issues/15322

**`1.22.71 Chromium: 89.0.4389.114 (Release)`** | **`1.25.31 Chromium: 90.0.4430.72`**
-----------------------------------------------------|-------------------------------------------
Screenshot_20210421-225257_Brave | Screenshot_20210421-225530_Brave - Nightly

**Pull request merged**